dashboard/app/controllers/home_controller.rb

```diff
@@ -64,4 +84,29 @@ def terms_and_privacy
   def teacher_announcements
     render template: 'api/teacher_announcement', layout: false
   end
+
+  def recent_courses
+    current_user.in_progress_and_completed_scripts.map do |script|
+      script_id = script[:script_id]
+      script_name = Script.find_by_id(script_id)[:name]
+      {
+        id: script_id,
+        script_name: script_name,
+        courseName: data_t_suffix('script.name', script_name, 'title'),
+        description: data_t_suffix('script.name', script_name, 'description_short'),
+        link: script_url(script_id),
+        image: "",
+        assignedSections: []
+      }
+    end
+  end
+
+  def courses
+    if current_user
+      @recent_courses = recent_courses
```

Comment:
To get the data you need, are you calling a Helper?  

I had guessed Rails' Controller uses these instance variables because it sometimes explicitly renders the HTML itself whereas other times it relies on the default routing to implicitly render that view.  But if you wanted to get the values more explicitly, you could use a Rails Helper instead.